pkg/flow/temporary.go

```diff
@@ -372,19 +372,35 @@ func (child *subflowHandle) Info() states.ChildInfo {
 
 func (engine *engine) newIsolateRequest(ctx context.Context, im *instanceMemory, stateId string, timeout int,
 	fn model.FunctionDefinition, inputData []byte,
-	uid uuid.UUID, async bool, files []model.FunctionFileDefinition, iterator int,
+	uid uuid.UUID, async bool, files []model.FunctionFileDefinition, branch int,
 ) (*functionRequest, error) {
 	ar := new(functionRequest)
 	ar.ActionID = uid.String()
-	ar.Workflow.Timeout = timeout
-	ar.Workflow.NamespaceName = im.instance.TelemetryInfo.NamespaceName
-	ar.Workflow.Path = im.instance.Instance.WorkflowPath
-	ar.Iterator = iterator
-	if !async {
-		ar.Workflow.InstanceID = im.ID().String()
-		ar.Workflow.NamespaceID = im.instance.Instance.NamespaceID.String()
-		ar.Workflow.State = stateId
-		ar.Workflow.Step = im.Step()
+
+	fnDescent := make([]enginerefactor.ParentInfo, 0, len(im.instance.DescentInfo.Descent)+1)
+	fnDescent = append(fnDescent, im.instance.DescentInfo.Descent...)
+	fnDescent = append(fnDescent, enginerefactor.ParentInfo{
+		ID:     im.GetInstanceID(),
+		State:  im.GetState(),
+		Step:   im.Step(),
+		Branch: branch,
+	})
+
+	ar.fnContext = enginerefactor.FunctionContext{
+		InstanceID: im.GetInstanceID(),
+		Step:       im.Step(),
+		Branch:     branch,
+		State:      stateId,
+		Callers: enginerefactor.InstanceDescentInfo{
```

Comment:
Is this callers descent info necessary to pass to the function? I don't think so. We should remove it because HTTP headers have a character limit and this field is very unbounded.